### PR TITLE
CUDA BFloat16 and other improvements on abs

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Not sure if ROCm supports `std::abs` today, let's see the CI

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
Tests are passing, now ROCm supports `std::abs`.

File: aten/src/ATen/native/cuda/AbsKernel.cu [hunk 1]
```diff
@@ -27,10 +21,8 @@ __host__ __device__ static inline bool abs_wrapper(bool v) {
 
 void abs_kernel_cuda(TensorIterator& iter) {
   AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(ScalarType::Half, ScalarType::BFloat16, ScalarType::Bool, iter.dtype(), "abs_cuda", [&]() {
-    AT_SKIP_BFLOAT16_IF_NOT_ROCM(scalar_t, "abs_cuda", [&] {
-      gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
-        return abs_wrapper(a);
-      });
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return abs_wrapper(a);
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/AbsKernel.cu:25]: do you need abs_wrapper now, or can it just be std::abs here?

Comment (zasdfgbnm) [reply aten/src/ATen/native/cuda/AbsKernel.cu]: Looks like the special cast of bool and uint8_t is not needed as well.

Comment (zasdfgbnm) [reply aten/src/ATen/native/cuda/AbsKernel.cu]: I have updated this PR.

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**